### ISSUE TITLE
Preset Save - 'Save preset' button and Text box

### ISSUE THREAD

**Issue by dev3225** [amosproj/amos2022ws03-software-oscilloscope#118]
## User story
1. As a user
2. I want the current configuration of the oscilloscope display to be saved through a button 
3. So that I can access it in future for similar data visualization.

## Acceptance criteria
- [ ]  Button UI should be visible clearly to the user
- [ ]  Text box to enter name of preset.
- [ ]  User can click on the 'Save preset' button only after entering a preset name(eg. 'Preset 1').
- [ ]  After the user clicks 'Save preset' button, the setting should be saved so that it can be retrieved in future.

## Definition of done (DoD)
- [ ]  Feature/Bug/Change has been implemented
- [ ]  Application compiles successfully
- [ ]  Code is documented

- [ ]  Feature/Bug/Change is tested by at least one unit or e2e test
- [ ]  Tests have been passed without warnings (except "deprecated" warnings)

- [ ]  Changes have been reviewed
- [ ]  PR has been merged to dev branch

- [ ]  New dependencies have been added to bill of materials
- [ ]  Software architecture diagram has been updated
- [ ]  All acceptance criteria are fulfilled
- [ ]  Screenshot is attached to issue

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
